docs/features/optin.md

```diff
@@ -0,0 +1,104 @@
+```
+
+You should see output similar to the following:
+
+> {"severity":"info","timestamp":"2024-11-04T21:56:28.704Z","msg":"Running reconcile","controller":"sqlinstance-controller","controllerGroup":"sql.cnrm.cloud.google.com","controllerKind":"SQLInstance","SQLInstance":{"name":"my-sqlinstance","namespace":"default"},"namespace":"default","name":"my-sqlinstance","reconcileID":"577fe58d-14eb-4e5d-9642-c9ec4b1a3137","resource":{"name":"my-sqlinstance","namespace":"default"}}
+
+If the legacy controller is enabled (instead of the new direct controller), the
+logs will show "starting reconcile" for the resource instead of "Running reconcile".
+
+> {"severity":"info","timestamp":"2024-10-30T23:47:49.726Z","logger":"sqlinstance-controller","msg":"starting reconcile","resource":{"name":"my-sqlinstance","namespace":"default"}}
+
+
+## Applicability
+
+We recommend only enabling the experimental direct controller if it is
+necessary to work around a legacy controller bug or enable a new feature that
+is only available in the direct controller.
+
+If you're unsure if a particular resource has an experimental direct controller
+available and want to try out the new code wherever possible, read on.
+
+To find out if an experimental direct controller is available for a resource,
+check in the `apis/` directory for the service `*_types.go` file. If the file
+specifies either of the following legacy controller labels, then it is likely
+possible to enable the experimental direct controller for resources of that
+kind.
+
+* "cnrm.cloud.google.com/tf2crd=true"
+* "cnrm.cloud.google.com/dcl2crd=true"
+
+See [this example](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/f485d486553f1fc939eb8e487ccabaf2f2f032ed/apis/sql/v1beta1/instance_types.go#L598)
+of a `*_types.go` file that has the legacy controller label, which could
+indicate that there is an experimental direct controller available for
+resources of that particular kind.
+
+However, not all resources in the `apis/` directory with the legacy controller
+annotation will have an experimental direct controller available. This is
+because not all resources in the apis/ directory have a direct controller
+implemented yet. To confirm if there is an experimental direct controller
+available for one of the types in the apis/ dir, check to make sure there is
+an implementation of the controller for that type in `pkg/controller/direct`.
+The controllers are found in `*_controller.go` [[example]](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/pkg/controller/direct/sql/sqlinstance_controller.go).
+Also, verify that the direct controller is registered in `pkg/controller/direct/register/register.go`
+[[example]](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/f485d486553f1fc939eb8e487ccabaf2f2f032ed/pkg/controller/direct/register/register.go#L43).
+
+At this point, if there is both a type in the apis/ dir with a legacy controller
```

Comment:
Do we need this paragraph? To my mind, the common case is that the direct ctrl is enabled if we are a type in the apis dir for a resource or will soon be enabled as the resource is being converted.

Reply:
After modifying the wording based on @yuwenma's suggestion in https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/3091#discussion_r1828556840, it's definitely necessary to include this. Because now we are checking the CRDs (not apis/ dir), there are many CRDs with the legacy label that do not yet have a direct controller implemented.